### PR TITLE
CLOUDP-350358: Improve and increase crd2go test coverage

### PR DESCRIPTION
**Pull request by helderjs**
# Summary

## Checklist
- [ ] Have you linked a jira ticket and/or is the ticket in the title?
- [ ] Have you checked whether your jira ticket required DOCSP changes?
- [ ] Have you checked for release_note changes?
- [ ] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?

## Reminder (Please remove this when merging)
- Please try to Approve or Reject Changes the PR, keep PRs in review as short as possible
- Remember the following Communication Standards - use comment prefixes for clarity:
  * **blocking**: Must be addressed before approval.
  * **follow-up**: Can be addressed in a later PR or ticket.
  * **q**: Clarifying question.
  * **nit**: Non-blocking suggestions.
  * **note**: Side-note, non-actionable. Example: Praise 
  * --> no prefix is considered a question

### CODE REVIEW

File: tools/crd2go/internal/gotype/typedict.go [hunk 1]
```diff
@@ -149,7 +149,7 @@ func (td TypeDict) RenameType(parentNames []string, gt *GoType) error {
 	}
 	goType.Name = typeName
 	if goType.Name == "string" {
-		log.Printf("here")
+		log.Printf("here") //TODO handle this case better
 	}
```
Comment (josvazg) [inline tools/crd2go/internal/gotype/typedict.go]: This was a trace I missed to remove. We can remove it completely.
(This trace allowed me to use debugging on a particular type even when the code was passing here for many other non interesting cases)

File: tools/crd2go/pkg/crd2go/crd2go_test.go [hunk 2]
```diff
@@ -13,7 +13,7 @@
 // limitations under the License.
 //
 
-package crd2go_test
+package crd2go
```
Comment (josvazg) [inline tools/crd2go/pkg/crd2go/crd2go_test.go:16]: **q**: Why was this changed?

It is usually preferable to test packages from the outside, to focus on public interfaces. When testing from the inside it might be a code smell, as tests might be checking implementation details rather than public interfaces behaviour, or it might be due to some lack in the design that makes the code difficult to test or use. In this case tests were able to run from the outside without issues AFAIK.

Comment (helderjs) [reply tools/crd2go/pkg/crd2go/crd2go_test.go:16]: Sure, I believe this holds true if we want exclusively test the public interface of the package, and might be valid here once this is going to be a public (open source) package. However, the public interfaces implementations are usually a combination of several internal units, and testing these granular units directly, you can cover a vast number of edge cases with simple, focused tests.
Regarding `testing from the inside it might be a code smell, as tests might be checking implementation details`. The test only breaks if the fundamental contract of that small unit changes, which is precisely what a unit test should detect. The "brittle test" problem arises when people test how a function works, not what it accomplishes. This is poor test design, not an inherent flaw of white-box testing.
As last, I wanted all the tests to be under the same coverage.

Comment (josvazg) [reply tools/crd2go/pkg/crd2go/crd2go_test.go:16]: But in this case the tests worked fine without being in the same package and there is nothing added that required insider access.

Also it is perfectly fine in Go to have separate test files for testing from the outside and others for testing within. Here I see no need to change access.
